declarations cannot be removed since we need to support builds on releases before
r151041 (where illumos 14418 was introduced).
change the declaration so it works in both worlds.

see: https://www.illumos.org/issues/14418

diff -wpruN '--exclude=*.orig' a~/deps/v8/src/base/platform/platform-posix.cc a/deps/v8/src/base/platform/platform-posix.cc
--- a~/deps/v8/src/base/platform/platform-posix.cc	1970-01-01 00:00:00
+++ a/deps/v8/src/base/platform/platform-posix.cc	1970-01-01 00:00:00
@@ -70,9 +70,9 @@
 
 #if defined(V8_OS_SOLARIS)
 #if (defined(_POSIX_C_SOURCE) && _POSIX_C_SOURCE > 2) || defined(__EXTENSIONS__)
-extern "C" int madvise(caddr_t, size_t, int);
+extern "C" int madvise(void *, size_t, int);
 #else
-extern int madvise(caddr_t, size_t, int);
+extern int madvise(void *, size_t, int);
 #endif
 #endif
 
